packages/next/server/next-server.ts

```diff
@@ -1653,6 +1703,54 @@ export default class Server {
           query: origQuery,
         })
 
+        const isFlight = pathname.endsWith('.server')
+        if (isFlight) {
+          // TODO: unstable_getCacheForType
+          const cache = new Map<string, any>()
+
+          let done = false
+          const flightResponse = this.flightWorker.render(
+            this.distDir,
+            pathname,
+            query
+          )
```

Comment:
I was hoping you would expose a `renderFlight` function to `RenderOpts` and move all of the RSC handling to `render` in `server/render.tsx`.